### PR TITLE
feat: DIA-977: Get streaming job status 

### PR DESCRIPTION
**Pull request by hakan458**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -49,6 +49,7 @@ services:
       - AWS_SESSION_TOKEN=${AWS_SESSION_TOKEN}
       - MODULE_NAME=process_file.app
       - KAFKA_BOOTSTRAP_SERVERS=kafka:9093
+      - C_FORCE_ROOT=true
```
Comment (pakelley) [inline docker-compose.yml:52]: why do we need this?

File: server/tasks/process_file.py [hunk 2]
```diff
@@ -30,32 +34,82 @@ def process_file(serialized_agent: bytes):
...
 
 
-@app.task(name="process_file_streaming", track_started=True, bind=True)
-def process_file_streaming(self, serialized_agent: bytes):
-    # Load the agent
-    agent = pickle.loads(serialized_agent)
+@app.task(
+    name="streaming_parent_task", track_started=True, bind=True, serializer="pickle"
+)
+def streaming_parent_task(
```
Comment (pakelley) [inline server/tasks/process_file.py:40]: nit: can we name this something like `streaming_coordinator_task` or something, to make the responsibility of the task more clear?

File: server/tasks/process_file.py [hunk 3]
```diff
@@ -30,32 +34,82 @@ def process_file(serialized_agent: bytes):
...
+    self, agent: Agent, result_handler: ResultHandler, batch_size: int = 2
+):
+    # Parent job ID is used for input/output topic names
+    parent_job_id = self.request.id
+
+    task = process_file_streaming
```
Comment (pakelley) [inline server/tasks/process_file.py:46]: nit: I know we just overwrite this later, but can we just name this `inference_task` or something? Also, in general, I think "inference" is more descriptive here than "input". If it seems clear to you too, imo we should just rename these vars to make the flow more clear.

File: server/tasks/process_file.py [hunk 4]
```diff
@@ -30,32 +34,82 @@ def process_file(serialized_agent: bytes):
...
+    logger.info(f"Submitting task {task.name} with agent {agent}")
+    input_result = task.delay(agent=agent, parent_job_id=parent_job_id)
+    input_job_id = input_result.id
+    logger.info(f"Task {task.name} submitted with job_id {input_job_id}")
+
+    task = process_streaming_output
```
Comment (pakelley) [inline server/tasks/process_file.py:52]: Similar here, I think `relay_task` or `result_handler_task` would be more descriptive than `output_task`

File: server/tasks/process_file.py [hunk 5]
```diff
@@ -30,32 +34,82 @@ def process_file(serialized_agent: bytes):
...
+    input_job = process_file_streaming.AsyncResult(input_job_id)
+    output_job = process_streaming_output.AsyncResult(output_job_id)
+
+    terminal_statuses = ["SUCCESS", "FAILURE", "REVOKED"]
+
+    while (
+        input_job.status not in terminal_statuses
+        or output_job.status not in terminal_statuses
+    ):
+        time.sleep(1)
```
Comment (pakelley) [inline server/tasks/process_file.py:79]: I'm pretty sure this is going to be blocking; I can't remember quite how this works in python, but is there a simple way to release the CPU while we're waiting here?

File: server/tasks/process_file.py [hunk 6]
```diff
@@ -30,32 +34,82 @@ def process_file(serialized_agent: bytes):
...
     # Run the agent
     asyncio.run(agent.arun())
 
 
 async def async_process_streaming_output(
-    input_job_id: str, result_handler: str, batch_size: int
+    input_job_id: str,
+    parent_job_id: str,
+    result_handler: ResultHandler,
+    batch_size: int,
```
Comment (pakelley) [inline server/tasks/process_file.py:108]: Would it make sense to provide a default `batch_size`?

**Review by pakelley — COMMENTED** [review]
I know it's a little tough to unit test this flow, but if you can find anything we could cover, it would be nice to at least get something here covered

File: docker-compose.yml (hunk 1 above)
Comment (hakan458) [reply docker-compose.yml:52]: when using pickle serializer in celery, and running celery as root in the docker container, this needs to be set. Going to check to make sure though that our real deployments arent running celery as root with plate team

File: server/tasks/process_file.py (hunk 2 above)
Comment (hakan458) [reply server/tasks/process_file.py:40]: Hmm this is pretty clear to me IMO haha

File: server/tasks/process_file.py (hunk 4 above)
Comment (hakan458) [reply server/tasks/process_file.py:52]: ? its not called `output_task` but we can change it I guess

Comment (hakan458) [reply server/tasks/process_file.py:52]: Ooo the left hand side you mean, ok

File: server/tasks/process_file.py (hunk 3 above)
Comment (hakan458) [reply server/tasks/process_file.py:46]: If you mean the left hand side `task` then yes I understand, agreed

File: server/tasks/process_file.py (hunk 5 above)
Comment (hakan458) [reply server/tasks/process_file.py:79]: No time.sleep does release the GIL - not blocking. This is the simple way

File: server/tasks/process_file.py (hunk 6 above)
Comment (hakan458) [reply server/tasks/process_file.py:108]: yes and it should be coming from `process_streaming_output` which was there, will add back

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/d22e9da72605cb06787cdc422de58ff3f46c955b is created.

File: docker-compose.yml (hunk 1 above)
Comment (pakelley) [reply docker-compose.yml:52]: gotcha. Mind adding a comment to this effect, so we can know if/when we can remove it?

File: server/tasks/process_file.py (hunk 2 above)
Comment (pakelley) [reply server/tasks/process_file.py:40]: not the hill I'll die on, but I could imagine a new dev having questions like "why is there a parent job?" and "what is this 'parent job' actually doing?".
I also like that in our diagrams we can have a more meaningful name for these.
That said, "coordinator" isn't too much more descriptive than "parent job", so if this isn't convincing that's fine.

Comment (hakan458) [reply server/tasks/process_file.py:40]: Id rather leave a comment on what its doing, than change the name to what I think isnt any more obvious really so Ill do that

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/4bbae560de56bf92b31a4f17e392f8857dade975 is created.

**Review by pakelley — APPROVED** [review]

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/75e8b64197e699050740b33d63b68648a768d125 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/a78e6d7bb3a61122ce15285f31ad521b1ed17785 is created.

File: server/app.py [hunk 7]
```diff
@@ -284,6 +275,68 @@ def get_status(job_id):
...
+        return "FAILURE"
+    if "REVOKED" in statuses:
+        return "REVOKED"
+    if "STARTED" in statuses or "RETRY" in statuses:
+        return "STARTED"
+
+    return "SUCCESS"
+
+
+@app.get("/streaming-jobs/{job_id}", response_model=Response[JobStatusResponse])
```
Comment (forum-hs) [inline server/app.py:296]: can we use jobs instead of streaming-jobs to be consistent ?

Comment (forum-hs) [reply server/app.py:296]: we are not going to support s3 anyways

**Review by forum-hs — CHANGES_REQUESTED** [review]
added comment about api path

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/bd09d4f80b7a2c6f0203c772d196cd645ba37c08 is created.

File: server/app.py (hunk 7 above)
Comment (matt-bernstein) [reply server/app.py:296]: also should update the endpoint touched by GET modelrun in LSE when streaming flag is on

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/6c8ff30b4b6e7900b2e0d1b45130c05fa320039a is created.

File: docker-compose.yml [hunk 8]
```diff
@@ -48,6 +48,7 @@ services:
       - AWS_SESSION_TOKEN=${AWS_SESSION_TOKEN}
       - MODULE_NAME=process_file.app
       - KAFKA_BOOTSTRAP_SERVERS=kafka:9093
+      - C_FORCE_ROOT=true # needed when using pickle serializer in celery + running as root - remove when we dont run as root
```
Comment (matt-bernstein) [inline docker-compose.yml:51]: weird, I thought this was already added

**Review by matt-bernstein — CHANGES_REQUESTED** [review]

File: server/tasks/process_file.py [hunk 9]
```diff
@@ -33,25 +36,92 @@ def process_file(agent: Agent):
     asyncio.run(agent.environment.finalize())
 
 
+@app.task(
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (hakan458) [reply server/tasks/process_file.py:40]: wtf I pushed this fix

File: server/tasks/process_file.py [hunk 10]
```diff
@@ -33,25 +36,92 @@ def process_file(agent: Agent):
...
+
+    logger.info("Both input and output jobs complete")
+
+    # Update parent task status to SUCCESS and pass metadata again
+    # otherwise its overwritten to None
+    self.update_state(
+        state=states.SUCCESS,
+        meta={"input_job_id": input_job_id, "output_job_id": output_job_id},
+    )
+    raise Ignore()
```
[the remark this answers is not in the document]
Comment (hakan458) [reply server/tasks/process_file.py:100]: No, it doesnt. I will leave comment

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/30258909e79194c78b04310fa32e8fad3d9cf9b1 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/e72f059f3ebca6d6c95ce735150ad50542a55d9f is created.

**Review by matt-bernstein — APPROVED** [review]

**Review by forum-hs — APPROVED** [review]

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/574869b18510ee47a51800faeef141d8cb2d2f9c is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/d361e5b858c876dae06d39441d687ebffaa96cf6 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/cee4ede219234ac555d94c7459fe3503ac958f79 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/c72ebcf5ba4edf8faa78ed763b0098a0a2a559ce is created.

**Pull request merged**